src/libtiled/tilelayer.h

```diff
@@ -349,7 +349,9 @@ class TILEDSHARED_EXPORT TileLayer : public Layer
     /**
      * Returns the bounds of this layer.
      */
-    QRect bounds() const { return QRect(mX, mY, mWidth, mHeight); }
+    QRect bounds() const { return mBounds.translated(mX, mY); }
```

Comment:
I suggested to use `bounds` in the map renderers, but due to an assert this will currently only work if this function is implemented as:
```c++
QRect bounds() const { return mBounds.translated(mX, mY).intersected(rect()); }
```
That is because `mBounds` may extend beyond the limits imposed by the layer size due to growing it chunk-by-chunk. I think that is a fine implementation for now, and the intersection can be taken out again when we actually allow layers to grow (and take out the assert in `cellAt`).